### PR TITLE
Fix bug around routing intent

### PR DESCRIPTION
**Pull request by MilesCameron-DMs**
# Overview/Summary

Please see https://github.com/Azure/bicep-lz-vending/issues/54

## This PR fixes/adds/changes/removes

1. Passes an object over for routing config when it should be null

### Breaking Changes

None

## Testing Evidence

I have tested this with routing intent enabled in our test and production environment.

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/bicep-lz-vending/wiki/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/bicep-lz-vending/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/bicep-lz-vending/issues)
- [ ] *(LZ-Vending Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/lz-vending/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/bicep-lz-vending/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/bicep-lz-vending/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/bicep-lz-vending/tree/main/.github/workflows/code-review.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)

### CODE REVIEW

File: main.bicep [hunk 1]
```diff
@@ -383,6 +383,15 @@ param virtualNetworkVwanPropagatedRouteTablesResourceIds array = []
 ''')
 param virtualNetworkVwanPropagatedLabels array = []
 
+@metadata({
+  example: false
+})
+@sys.description('''Indicates whether routing intent is enabled on the Virtual HUB within the virtual WAN.
```
Comment (jtracey93) [inline main.bicep:389]:
```suggestion
@sys.description('''Indicates whether routing intent is enabled on the Virtual Hub within the Virtual WAN.
```

File: main.bicep.parameters.md [hunk 2]
```diff
@@ -40,6 +40,7 @@ virtualNetworkVwanEnableInternetSecurity | No       | Enables the ability for th
...
 virtualNetworkVwanPropagatedRouteTablesResourceIds | No       | An array of of objects of virtual hub route table resource IDs to propagate routes to. If left blank/empty the `defaultRouteTable` will be propagated to only.  Each object must contain the following `key`: - `id` = The Resource ID of the Virtual WAN Virtual Hub Route Table IDs you wish to propagate too  > See below [example in parameter file](#parameter-file)  > **IMPORTANT:** If you provide any Route Tables in this array of objects you must ensure you include also the `defaultRouteTable` Resource ID as an object in the array as it is not added by default when a value is provided for this parameter.  - Type: `[]` Array - Default value: `[]` *(empty array)* 
 virtualNetworkVwanPropagatedLabels | No       | An array of virtual hub route table labels to propagate routes to. If left blank/empty the default label will be propagated to only.  - Type: `[]` Array - Default value: `[]` *(empty array)* 
+vHubRoutingIntentEnabled | No       | Indicates whether routing intent is enabled on the Virtual HUB within the virtual WAN.  - Type: Boolean
```
Comment (jtracey93) [inline main.bicep.parameters.md:43]:
```suggestion
vHubRoutingIntentEnabled | No       | Indicates whether routing intent is enabled on the Virtual Hub within the Virtual WAN.  - Type: Boolean
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Minor typos

**Review by MilesCameron-DMs — COMMENTED** [review]
Looks good to me @jtracey93

**Comment by jtracey93** [comment]
Tests are failing due to #56, will resolve in a later PR

**Review by jtracey93 — APPROVED** [review]
LGTM

**Comment by jtracey93** [comment]
FYI @MilesCameron-DMs we will need to cut a release on the upstream Bicep Public Registry, so this wont be available instantly on the registry 👍

**Pull request merged**